src/test/java/com/iota/iri/service/validation/impl/TransactionSolidifierImplTest.java

```diff
@@ -0,0 +1,149 @@
+import com.iota.iri.crypto.SpongeFactory;
+import com.iota.iri.model.TransactionHash;
+import com.iota.iri.network.TransactionRequester;
+import com.iota.iri.service.snapshot.SnapshotProvider;
+import com.iota.iri.service.snapshot.impl.SnapshotMockUtils;
+import com.iota.iri.storage.Tangle;
+import com.iota.iri.storage.rocksDB.RocksDBPersistenceProvider;
+import com.iota.iri.utils.Converter;
+import org.junit.Rule;
+import org.junit.BeforeClass;
+import org.junit.Before;
+import org.junit.AfterClass;
+import org.junit.After;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnit;
+import org.mockito.junit.MockitoRule;
+
+import static com.iota.iri.TransactionTestUtils.getTransactionTrits;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+import static org.mockito.Mockito.when;
+
+public class TransactionSolidifierImplTest {
+    private static final TemporaryFolder dbFolder = new TemporaryFolder();
+    private static final TemporaryFolder logFolder = new TemporaryFolder();
+    private static Tangle tangle;
+
+    @Rule
+    public MockitoRule mockitoRule = MockitoJUnit.rule();
+
+    @Mock
+    private static SnapshotProvider snapshotProvider;
+
+    @Mock
+    private static TransactionSolidifierImpl txSolidifier;
+
+    @Mock
+    private static TransactionRequester txRequester;
+
+    @BeforeClass
+    public static void setUp() throws Exception {
+        dbFolder.create();
+        logFolder.create();
+        tangle = new Tangle();
+        tangle.addPersistenceProvider(
+                new RocksDBPersistenceProvider(
+                        dbFolder.getRoot().getAbsolutePath(), logFolder.getRoot().getAbsolutePath(),1000, Tangle.COLUMN_FAMILIES, Tangle.METADATA_COLUMN_FAMILY));
+        tangle.init();
+    }
+
+    @AfterClass
+    public static void tearDown() throws Exception {
+        tangle.shutdown();
+        dbFolder.delete();
+        logFolder.delete();
+    }
+
+    @Before
+    public void setUpEach() {
+        when(snapshotProvider.getInitialSnapshot()).thenReturn(SnapshotMockUtils.createSnapshot());
+        txRequester = new TransactionRequester(tangle, snapshotProvider);
+        txSolidifier = new TransactionSolidifierImpl(tangle, snapshotProvider, txRequester);
+        txSolidifier.start();
+    }
+
+    @After
+    public void tearDownEach(){
+        txSolidifier.shutdown();
+    }
+
+
+    @Test
+    public void verifyTxIsSolid() throws Exception {
+        TransactionViewModel tx = getTxWithBranchAndTrunk();
+        assertTrue(txSolidifier.checkSolidity(tx.getHash()));
+        assertTrue(txSolidifier.checkSolidity(tx.getHash()));
+    }
+
+    @Test
+    public void verifyTxIsNotSolid() throws Exception {
+        TransactionViewModel tx = getTxWithoutBranchAndTrunk();
+        assertFalse(txSolidifier.checkSolidity(tx.getHash()));
+        assertFalse(txSolidifier.checkSolidity(tx.getHash()));
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [JUnit assertions should include a message](https://app.codacy.com/app/GalRogozinski/iri/pullRequest?prid=4339340)